pkg/datapath/linux/config/config.go

```diff
@@ -525,7 +526,12 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 	}
 
 	if option.Config.EnableIPSec {
-		a := byteorder.NetIPv4ToHost32(node.GetIPv4())
+		nodeAddress := node.GetIPv4()
+		if nodeAddress == nil {
+			return errors.New("external IPv4 node address is required when IPSec is enabled, but none found")
```

Comment:
Shouldn't we fail even earlier, during the agent bootstrap, with a fatal if we are running IPv6-only? See all the fatals in daemon_main.go for example.

Reply:
Currently, the same check concerning vxlan tunneling (and, wireguard support -- #23552), is performed in the [node.ValidatePostInit function](https://github.com/cilium/cilium/blob/705afced96563f8fefe448e9424ee2dcf5a24b5d/pkg/node/address.go#L554-L558), since it requires that the node IPv4 address has already been retrieved (if any). Still, adding the check there is too late in this case, since the generation of the header file is performed earlier (in `newDaemon`, while the check is part of `runDaemon` [[1]](https://github.com/cilium/cilium/blob/705afced96563f8fefe448e9424ee2dcf5a24b5d/daemon/cmd/daemon_main.go#L1676-L1711)).

In my opinion, having the check based only on the flags configuration could lead to issues, since the IPv4 address might not be present also due to wrong configuration, while AFAIU it might be present even if cilium is configured in IPv6-only mode (and in that case IPsec should work correctly).

Reply:
> In my opinion, having the check based only on the flags configuration could lead to issues, since the IPv4 address might not be present also due to wrong configuration

Hm, that's right. We need two checks then :grin:

Reply:
I assume you mean preventing cilium to start if IPsec is enabled, and IPv4 is disabled (correct me if I'm not getting it correctly). Wouldn't that risk to disable this feature even if it could work (i.e., the nodes have an IPv4 address)?

Reply:
I'm dubious that it can work if the node supports IPv4 but you disabled it in Cilium. I suspect other things would then be disabled and prevent this.

In any case, it seems very corner case. I'd rather disable it and actual users can always come complain later.